### PR TITLE
haskell-language-server: fix test, prepare ARM build

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

The current formula encodes the full version of GHC in the name of the
binary. This leads to a test failure in GHC version bump PRs (cf. #78821).
Let's fix that by using only the major and minor versions.

While we're here, let's get this formula ready for bottling for ARM.

CC @kret, since this renames the binaries from `haskell-language-server-8.10.4` to `haskell-language-server-8.10`, etc. I hope that won't break anything.

### CODE REVIEW

File: Formula/haskell-language-server.rb [hunk 1]
```diff
@@ -63,11 +76,11 @@ def caveats
       f :: Int -> Int
     EOS
 
-    %w[ghc@8.6 ghc@8.8 ghc].each do |ghc_str|
-      ghc = Formula[ghc_str]
-
+    ghcs.each do |ghc|
       assert_match "Completed (1 file worked, 1 file failed)",
-        shell_output("PATH=#{ghc.bin}:$PATH #{bin}/haskell-language-server-#{ghc.version} #{testpath}/*.hs 2>&1", 1)
+        with_env(PATH: "#{ghc.bin}:$PATH") do
```
Comment (carlocab) [inline Formula/haskell-language-server.rb:81]: Oops. This should probably be
```suggestion
        with_env(PATH: "#{ghc.bin}:#{ENV["PATH"]}") do
```
Will fix this in a bit.

**Comment by kret** [comment]
So, renaming binaries as you propose will not break HLS.
IDEs point to wrapper binary, and it looks for:
1. `haskell-language-server-x.y.z`
2. `haskell-language-server-x.y`
3. `haskell-language-server-x`
4. `haskell-language-server`

binaries in the PATH, in this order.

I used `x.y.z` because this is how HLS releases their binaries, and also because it is immediately clear what version of GHC the HLS is built for. Of course it is still possible to query that by `--version` so it is not a blocker for anyone, I guess.

File: Formula/haskell-language-server.rb [hunk 2]
```diff
@@ -63,11 +78,11 @@ def caveats
...
 
-    %w[ghc@8.6 ghc@8.8 ghc].each do |ghc_str|
-      ghc = Formula[ghc_str]
-
+    ghcs.each do |ghc|
       assert_match "Completed (1 file worked, 1 file failed)",
-        shell_output("PATH=#{ghc.bin}:$PATH #{bin}/haskell-language-server-#{ghc.version} #{testpath}/*.hs 2>&1", 1)
+        with_env(PATH: "#{ghc.bin}:#{ENV["PATH"]}") do
+          shell_output("#{bin}/haskell-language-server-#{ghc.version.major_minor} #{testpath}/*.hs 2>&1", 1)
+        end
```
Comment (cho-m) [inline Formula/haskell-language-server.rb:85]: I don't think `with_env` block will return anything.

If this is wanted, then may need to use a variable
```rb
ghcs.each do |ghc|
  output = nil
  with_env(...) do
    output = shell_output(...)
  end
  assert_match "Completed (1 file worked, 1 file failed)", output
end

Comment (carlocab) [reply Formula/haskell-language-server.rb:85]: Good catch. Will fix this.

File: Formula/haskell-language-server.rb [hunk 3]
```diff
@@ -20,29 +21,43 @@ class HaskellLanguageServer < Formula
...
+      depends_on "ghc@8.6" => [:build, :test]
+      depends_on "ghc@8.8" => [:build, :test]
+    end
+  end
+
+  on_linux do
+    depends_on "ghc" => [:build, :test]
+    depends_on "ghc@8.6" => [:build, :test]
+    depends_on "ghc@8.8" => [:build, :test]
+  end
```
Comment (cho-m) [inline Formula/haskell-language-server.rb:39]: Do we need a separate linux section anymore?
Based on other Formulae, I don't think we usually care about Linux ARM as a use-case.
```rb
  depends_on "ghc" => [:build, :test]
  depends_on "llvm" => :build if Hardware::CPU.arm?
  unless Hardware::CPU.arm?
    depends_on "ghc@8.6" => [:build, :test]
    depends_on "ghc@8.8" => [:build, :test]
  end
```

**Pull request closed without merging**